activerecord/lib/active_record/autosave_association.rb

```diff
@@ -456,9 +456,18 @@ def save_has_one_association(reflection)
 
       # If the record is new or it has changed, returns true.
       def record_changed?(reflection, record, key)
-        record.new_record? ||
-          (record.has_attribute?(reflection.foreign_key) && record[reflection.foreign_key] != key) ||
-          record.will_save_change_to_attribute?(reflection.foreign_key)
+        return true if record.new_record?
+        return true if association_key_changed?(reflection, record, key)
+        return true if record.will_save_change_to_attribute?(reflection.foreign_key)
+
+        false
+      end
+
+      # If association foreign key was changed
+      def association_key_changed?(reflection, record, key)
```

Comment:
Instead of leaving a comment can you update the method name to be `association_foreign_key_changed?` and then move this to a private namespace since we don't want anyone using this directly in their applications.

Reply:
good point. as for the private, it's already inside the private space, but I added `_` prefix to it, to mark the name as private too

Reply:
Sorry didn't realize it was already private - you can leave it without the `_` prefix then.

Reply:
I saw a method below with underscore, so assume that it's ok too. I would leave it as is